bokehjs/src/lib/models/tools/tool.ts

```diff
@@ -90,6 +91,12 @@ export abstract class ToolView extends View {
   // deactivate is triggered by toolbar ui actions
   deactivate(): void {}
 
+  // this is just here to avoid duplicating it on many various tools
+  _trigger_ranges_update(): void {
+    const {x_range, y_range} = this.plot_model
+    this.plot_model.trigger_event(new RangesUpdate(x_range.start, x_range.end, y_range.start, y_range.end))
+  }
```

Comment:
This is what class structure is for. You are factoring common code out. This comment is redundant.

Reply:
If fact this method doesn't belong here, as it doesn't manipulate a `Tool` in any way.

Reply:
Hence the comment, it's not ideal, but it is better than repeating it copy-pasted in 5 places